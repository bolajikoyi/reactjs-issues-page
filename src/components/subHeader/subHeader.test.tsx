import React from 'react';
import { render, screen, cleanup } from '@testing-library/react';
import ReactDOM from 'react-dom';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';
import SubHeader from './subHeader';

afterEach(cleanup);

let data = {
    totalCount: 100,
    isClosed: true
}

describe('SubHeader Component', ()=> {
    test('Expect componet to be created', async () => {
        render(<SubHeader count={data}/>);
        expect(SubHeader).toBeTruthy();
    });

    test('check if Code, Issues, Pull Request, Actions, Security, and Insights are in the document', ()=> {
        render(<SubHeader count={data}/>);
        expect(screen.getByTestId('code')).toBeInTheDocument();
        expect(screen.getByTestId('issues')).toBeInTheDocument();
        expect(screen.getByTestId('pull-requests')).toBeInTheDocument();
        expect(screen.getByTestId('actions')).toBeInTheDocument();
        expect(screen.getByTestId('security')).toBeInTheDocument();
        expect(screen.getByTestId('insights')).toBeInTheDocument();
    })

    test('matches snapshot', ()=> {
        const tree = renderer.create(<SubHeader count={data}/>).toJSON();
        expect(tree).toMatchSnapshot();
    })
})

