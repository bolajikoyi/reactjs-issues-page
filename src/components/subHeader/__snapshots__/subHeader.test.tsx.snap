// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubHeader Component matches snapshot 1`] = `
<div
  className="SubHeader"
>
  <div
    className="SubHeader-top"
  >
    <div
      className="left"
    >
      <span>
        <i
          className="subheader fas fa-clipboard"
        />
         
      </span>
      <span
        className="owner"
      >
        <a
          href="https://github.com/reactjs"
        >
          reactjs
        </a>
         
      </span>
      /
      <span
        className="repo-name"
      >
         
        <a
          href="https://github.com/reactjs/reactjs.org"
        >
          reactjs.org
        </a>
      </span>
      <span
        className="repo-visibility"
      >
         Public
      </span>
    </div>
    <div
      className="right"
    >
      <div
        className="watch"
      >
        <span>
          <i
            className="far fa-eye"
          />
        </span>
        <span>
          Watch
        </span>
        <span
          className="number"
        >
          181
        </span>
        <span>
          <i
            className="fas fa-caret-down"
          />
        </span>
      </div>
      <div
        className="fork"
      >
        <span>
          <img
            alt="github fork icon"
            className="fork-icon"
            src="fork.png"
          />
        </span>
        <span>
          Fork
        </span>
        <span
          className="number"
        >
          5.7k
        </span>
      </div>
      <div
        className="star"
      >
        <span>
          <i
            className="far fa-star"
          />
        </span>
        <span>
          Star
        </span>
        <span
          className="number"
        >
          6.8k
        </span>
        <span>
          <i
            className="fas fa-caret-down"
          />
        </span>
      </div>
    </div>
  </div>
  <div
    className="SubHeader-bottom"
  >
    <div
      className="code"
    >
      <span
        data-testid="code"
      >
        <i
          className="subheader fas fa-code"
        />
         Code 
      </span>
    </div>
    <div
      className="issues"
    >
      <span
        data-testid="issues"
      >
        <i
          className="far fa-check-circle"
        />
         Issues 
        <span
          className="number"
        >
          100
        </span>
      </span>
    </div>
    <div
      className="pull-request"
    >
      <span
        data-testid="pull-requests"
      >
        <img
          alt="pull request icon"
          className="subheader pull-request-icon"
          src="pull-request.png"
        />
         Pull requests 
        <span
          className="number"
        >
          584
        </span>
      </span>
    </div>
    <div
      className="actions"
    >
      <span
        data-testid="actions"
      >
        <i
          className="subheader far fa-play-circle"
        />
         Actions
      </span>
    </div>
    <div
      className="security"
    >
      <span
        data-testid="security"
      >
        <i
          className="subheader fas fa-shield-alt"
        />
         Security
      </span>
    </div>
    <div
      className="insight"
    >
      <span
        data-testid="insights"
      >
        <i
          className="subheader fas fa-chart-line"
        />
         Insights
      </span>
    </div>
  </div>
</div>
`;
